### PR TITLE
Features/draft

### PR DESCRIPTION
**Pull request by mykola-strongsd**

### CODE REVIEW

File: lib/adhoq/storage/cache.rb [hunk 1]
```diff
@@ -22,10 +22,6 @@ def direct_download?
       def get(id)
         cache.read(cache_prefix + id)
       end
-
-      def get_url(_report)
-        raise NotImplementedError
-      end
```
Comment (romikoops) [inline lib/adhoq/storage/cache.rb]: Why did you remove it?

File: spec/models/adhoq/execution_spec.rb [hunk 2]
```diff
@@ -22,6 +26,40 @@ module Adhoq
       expect(execution.report.data).to be_nil
     end
 
+    context 'when report file name prefix is specified' do
+      before do
+        Adhoq.config.report_file_name_prefix = 'prefix_test_'
```
Comment (romikoops) [inline spec/models/adhoq/execution_spec.rb:31]: It will affect all test suite

**Review by romikoops — COMMENTED** [review]
Please see my comments

File: lib/adhoq/storage/cache.rb (hunk 1 above)
Comment (mykola-strongsd) [reply lib/adhoq/storage/cache.rb]: Moved to base class - `lib/adhoq/storage/fog_storage.rb `

File: spec/models/adhoq/execution_spec.rb (hunk 2 above)
Comment (mykola-strongsd) [reply spec/models/adhoq/execution_spec.rb:31]: fixed by adding `after do...end`

File: lib/adhoq/storage/cache.rb (hunk 1 above)
Comment (romikoops) [reply lib/adhoq/storage/cache.rb]: https://github.com/romikoops/adhoq/blob/master/lib/adhoq/storage/cache.rb#L3 Base class is Object

Comment (mykola-strongsd) [reply lib/adhoq/storage/cache.rb]: fixed by "revert back get_url to Storage::Cache and Storage::OnTheFly"
